Content.Shared/Dataset/LocalizedDatasetPrototype.cs

```diff
@@ -0,0 +1,94 @@
+using System.Collections;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.Dataset;
+
+/// <summary>
+/// A variant of <see cref="DatasetPrototype"/> intended to specify a sequence of LocId strings
+/// without having to copy-paste a ton of LocId strings into the YAML.
+/// </summary>
+[Prototype("localizedDataset")]
+public sealed partial class LocalizedDatasetPrototype : IPrototype
+{
+    /// <summary>
+    /// Identifier for this prototype.
+    /// </summary>
+    [ViewVariables]
+    [IdDataField]
+    public string ID { get; private set; } = default!;
+
+    /// <summary>
+    /// Collection of LocId strings.
+    /// </summary>
+    [DataField]
+    public LocalizedDatasetValues Values { get; private set; } = [];
+}
+
+[Serializable, NetSerializable]
+[DataDefinition]
+public sealed partial class LocalizedDatasetValues : IReadOnlyList<string>
```

Comment:
i mean does it make that much of a difference to bother implementing IReadOnlyList? from a glance it looks like more code for little gain

Reply:
It's for parity with the normal DataSet which uses IReadOnlyList for its Values property. The main use case is `_random.Pick(dataset.Values)` which uses the Count and [] operator. The extension method makes that a bit redundant, but I've seen this format used more often. 

Being enumerable is probably a little excessive, but it might be useful for testing? But why not implement it? Is there a performance cost or something?